packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/billing_client_wrapper.dart

```diff
@@ -256,14 +218,24 @@ class BillingClient {
   /// This wraps
   /// [`BillingClient#queryPurchasesAsync(QueryPurchaseParams, PurchaseResponseListener)`](https://developer.android.com/reference/com/android/billingclient/api/BillingClient#queryPurchasesAsync(com.android.billingclient.api.QueryPurchasesParams,%20com.android.billingclient.api.PurchasesResponseListener)).
   Future<PurchasesResultWrapper> queryPurchases(ProductType productType) async {
-    return PurchasesResultWrapper.fromJson(
-        (await channel.invokeMapMethod<String, dynamic>(
-              'BillingClient#queryPurchasesAsync(QueryPurchaseParams, PurchaseResponseListener)',
-              <String, dynamic>{
-                'productType': const ProductTypeConverter().toJson(productType)
-              },
-            )) ??
-            <String, dynamic>{});
+    // TODO(stuartmorgan): Investigate whether forceOkResponseCode is actually
```

Comment:
@reidbaker @gmackall FYI, someone who can dig into the expected behavior should probably take a look at this ASAP. This is a Dart conversion of [this code](https://github.com/flutter/packages/pull/6262/files#diff-3c947ed9281a5b9d78b07931af2e897b70781f8c13769b7c638270e8fb263af0L479-L481), but for the reasons explained in this comment, I'm pretty skeptical that the Java code was correct. I deliberately didn't change it here since this PR is supposed to be a functional no-op, but I did want to flag it.

Reply:
Keep the code you have as is for this pr.